eth/fetcher/tx_fetcher.go

```diff
@@ -262,57 +262,82 @@ func (f *TxFetcher) Notify(peer string, hashes []common.Hash) error {
+	var (
+		inMeter, knownMeter, underpricedMeter, otherRejectMeter metrics.Meter
+	)
 	if direct {
-		txReplyInMeter.Mark(int64(len(txs)))
+		inMeter = txReplyInMeter
+		knownMeter = txReplyKnownMeter
+		underpricedMeter = txReplyUnderpricedMeter
+		otherRejectMeter = txReplyOtherRejectMeter
 	} else {
-		txBroadcastInMeter.Mark(int64(len(txs)))
+		inMeter = txBroadcastInMeter
+		knownMeter = txBroadcastKnownMeter
+		underpricedMeter = txBroadcastUnderpricedMeter
+		otherRejectMeter = txBroadcastOtherRejectMeter
 	}
+	// Keep track of all the propagated transactions
+	inMeter.Mark(int64(len(txs)))
+
 	// Push all the transactions into the pool, tracking underpriced ones to avoid
 	// re-requesting them and dropping the peer in case of malicious transfers.
 	var (
-		added       = make([]common.Hash, 0, len(txs))
-		duplicate   int64
-		underpriced int64
-		otherreject int64
+		added = make([]common.Hash, 0, len(txs))
+		delay time.Duration
 	)
-	errs := f.addTxs(txs)
-	for i, err := range errs {
-		// Track the transaction hash if the price is too low for us.
-		// Avoid re-request this transaction when we receive another
-		// announcement.
-		if errors.Is(err, core.ErrUnderpriced) || errors.Is(err, core.ErrReplaceUnderpriced) {
-			for f.underpriced.Cardinality() >= maxTxUnderpricedSetSize {
-				f.underpriced.Pop()
-			}
-			f.underpriced.Add(txs[i].Hash())
+	// proceed in batches
+	for i := 0; i < len(txs); i += 100 {
+		end := i + 100
+		if end > len(txs) {
+			end = len(txs)
 		}
-		// Track a few interesting failure types
-		switch {
-		case err == nil: // Noop, but need to handle to not count these
+		var (
+			duplicate   int64
+			underpriced int64
+			otherreject int64
+		)
+		batch := txs[i:end]
+		errs := f.addTxs(batch)
+		for j, err := range errs {
+			// Track the transaction hash if the price is too low for us.
+			// Avoid re-request this transaction when we receive another
+			// announcement.
+			if errors.Is(err, core.ErrUnderpriced) || errors.Is(err, core.ErrReplaceUnderpriced) {
+				for f.underpriced.Cardinality() >= maxTxUnderpricedSetSize {
+					f.underpriced.Pop()
+				}
+				f.underpriced.Add(batch[i].Hash())
+			}
+			// Track a few interesting failure types
+			switch {
+			case err == nil: // Noop, but need to handle to not count these
 
-		case errors.Is(err, core.ErrAlreadyKnown):
-			duplicate++
+			case errors.Is(err, core.ErrAlreadyKnown):
+				duplicate++
 
-		case errors.Is(err, core.ErrUnderpriced) || errors.Is(err, core.ErrReplaceUnderpriced):
-			underpriced++
+			case errors.Is(err, core.ErrUnderpriced) || errors.Is(err, core.ErrReplaceUnderpriced):
+				underpriced++
 
-		default:
-			otherreject++
+			default:
+				otherreject++
+			}
+			added = append(added, batch[j].Hash())
+		}
+		knownMeter.Mark(duplicate)
+		underpricedMeter.Mark(underpriced)
+		otherRejectMeter.Mark(otherreject)
+		// If 'other reject' is >25% of the deliveries in any batch, abort. Either we are
```

Comment:
Pls add an empty line before the commen, easyer on the eyes if there's a visual separator